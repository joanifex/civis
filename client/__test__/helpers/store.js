import configureMockStore from 'redux-mock-store';


const store = (model) => {
  const mockStore = configureMockStore();
  return mockStore(model)
}

export default store;
