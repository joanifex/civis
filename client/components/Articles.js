import React from 'react'

class Articles extends React.Component {

  displayArticles = () => {
    let { articles } = this.props
    return articles.map( (article, i) => {
      return(
        <li key={i} className="collection-item">
          <a href={article.web_url} target="_blank">
            {article.headline}
          </a>
          <p>{article.lead_paragraph}</p>
          <p>{article.snippet}</p>
          <p>{article.pub_date}</p>
        </li>
      );
    });
  }

  render(){
    return(
      <div>
        <h5>Articles</h5>
        <div className="row">
          <div className="col s12 m8">
            <ul className="collection">
              { this.displayArticles() }
            </ul>
          </div>
        </div>
      </div>
    );
  }
}

export default Articles;
