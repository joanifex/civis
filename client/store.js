import { createStore, compose, applyMiddleware } from 'redux';
import { syncHistoryWithStore } from 'react-router-redux';
import { browserHistory } from 'react-router';
import thunk from 'redux-thunk';

import rootReducer from './reducers/index';

const enhancers = compose(
  applyMiddleware(thunk),
  window.devToolsExtension ? window.devToolsExtension() : f => f
)

let appDataSet = document.getElementById('app').dataset;
let remember = appDataSet.remember;

remember = remember === 'true' ? true : false;

const store = createStore(rootReducer, { auth: { isAuthenticated: remember } }, enhancers)

if(module.hot) {
  module.hot.accept('./reducers/',() => {
    const nextRootReducer = require('./reducers/index').default;
    store.replaceReducer(nextRootReducer);
  });
}

export const history = syncHistoryWithStore(browserHistory, store)

export default store
