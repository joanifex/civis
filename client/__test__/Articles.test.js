import React from 'react';
import Articles from '../components/Articles.js';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from './helpers/store';

test('Component is rendered', () => {
  const component = renderer.create(
    <Provider store={store({})}>
      <Article />
   </Provider>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
