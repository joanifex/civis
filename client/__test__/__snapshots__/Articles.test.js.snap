// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component is rendered 1`] = `
<div>
  <h5>
    Articles
  </h5>
  <div
    className="row"
  >
    <div
      className="col s12 m8"
    >
      <ul
        className="collection"
      />
    </div>
  </div>
</div>
`;
